tests/suites/test_suite_mpi.function

```diff
@@ -880,3 +911,34 @@ void mpi_selftest(  )
     TEST_ASSERT( mbedtls_mpi_self_test( 1 ) == 0 );
 }
 /* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_GENPRIME */
+void mbedtls_mpi_bad_trng_check_test( data_t * input_X )
+{
+    mbedtls_mpi X;
+    int res;
+    int bits = 256;
+    mbedtls_test_mpi_random rand;
+
+    int flags_check = MBEDTLS_MPI_GEN_PRIME_CHECK_TRNG;
+
+    mbedtls_mpi_init( &X );
+
+    rand.data = input_X;
+    rand.pos = 0;
+    rand.chunk_len = 0;
+    res = mbedtls_mpi_gen_prime( &X, bits, flags_check, rnd_std_rand,
+                                 NULL);
+    TEST_ASSERT( res == 0 );
```

Comment:
I think it would be better to have separate test cases for the normal RNG with the `CHECK_TRNG` flag, and use this function only to test that `CHECK_TRNG` detects the made-up bad TRNG.